File: cawbs/Dart/lib/cawbs.dart
// Copyright (c) Core DF. All rights reserved.
//
// Documentation: https://coreauto.coredf.com/resources

import 'dart:io';
import 'wbs.dart';

final _sess = WbsSession();

String _env(String name) => Platform.environment[name] ?? '';

Future<WbsResult> init() async {
  final env = _env('ENV');
  final actionId = _env('ACTIONID');
  final accessCode = _env('CA_ACCESS_CODE');
  final baseUrl = _env('CA_WBS_URL');
  final stepName = _env('STEPNAME');
  if ([env, actionId, accessCode, baseUrl, stepName].any((v) => v.isEmpty)) {
    return WbsSession.missingEnv('ENV, ACTIONID, CA_ACCESS_CODE, CA_WBS_URL, STEPNAME');
  }
  return _sess.authenticate(env, accessCode, baseUrl);
}

Future<WbsResult> getEventPayload() => _sess.getEventPayload(_env('ACTIONID'));

Future<WbsResult> putStepPayload(dynamic payload) =>
    _sess.putStepPayload(_env('ACTIONID'), _env('STEPNAME'), payload);

Future<WbsResult> getStepPayload(String stepname) =>
    _sess.getStepPayload(_env('ACTIONID'), stepname);

Future<WbsResult> getKeystore(String keylist) => _sess.getKeystore(keylist);

File: cawbs/DotNet/Cawbs.cs
// Copyright (c) Core DF. All rights reserved.
//
// Core Auto Web Services library (cawbs) — .NET client for the Core Auto Collector.
//
// Documentation: https://coreauto.coredf.com/resources

namespace CoreAuto.Cawbs;

public static class Cawbs
{
    private static readonly WbsSession Sess = new();

    public static Task<Result> InitAsync()
    {
        var env = Environment.GetEnvironmentVariable("ENV");
        var actionId = Environment.GetEnvironmentVariable("ACTIONID");
        var accessCode = Environment.GetEnvironmentVariable("CA_ACCESS_CODE");
        var baseUrl = Environment.GetEnvironmentVariable("CA_WBS_URL");
        var stepName = Environment.GetEnvironmentVariable("STEPNAME");
        if (string.IsNullOrWhiteSpace(env) ||
            string.IsNullOrWhiteSpace(actionId) ||
            string.IsNullOrWhiteSpace(accessCode) ||
            string.IsNullOrWhiteSpace(baseUrl) ||
            string.IsNullOrWhiteSpace(stepName))
        {
            return Task.FromResult(WbsSession.MissingEnv("ENV, ACTIONID, CA_ACCESS_CODE, CA_WBS_URL, STEPNAME"));
        }
        return Sess.AuthenticateAsync(env, accessCode, baseUrl);
    }

    public static Task<Result> GetEventPayloadAsync() =>
        Sess.GetEventPayloadAsync(Environment.GetEnvironmentVariable("ACTIONID") ?? "");

    public static Task<Result> PutStepPayloadAsync(object payload) =>
        Sess.PutStepPayloadAsync(
            Environment.GetEnvironmentVariable("ACTIONID") ?? "",
            Environment.GetEnvironmentVariable("STEPNAME") ?? "",
            payload);

    public static Task<Result> GetStepPayloadAsync(string stepname) =>
        Sess.GetStepPayloadAsync(Environment.GetEnvironmentVariable("ACTIONID") ?? "", stepname);

    public static Task<Result> GetKeystoreAsync(string keylist) => Sess.GetKeystoreAsync(keylist);
}
